sdk/identity/identity/src/credentials/managedIdentityCredential/index.ts

```diff
@@ -241,7 +241,8 @@ export class ManagedIdentityCredential implements TokenCredential {
             scopes: Array.isArray(scopes) ? scopes : [scopes],
             claims: options?.claims,
           };
-
+          // Added a check to see if SetAppTokenProvider was already defined.
+          // Don't redefine it if it's already defined, since it should be static method.
```

Comment:
If you're removing the check you should remove the comment here. Is this not needlessly re-calling `SetAppTokenProvider`, though? Perhaps instead of checking it for false-ness, we just need a field noting it's initialized or something?

Reply:
`we just need a field noting it's initialized or something` - I like this idea, might work, will try it out!